feat: seoul recap blog post (#117)
gnolang/blog: posts/2024-11-28_seoul-retreat/README.md

---
publication_date: 2024-11-28T00:00:00Z
slug: seoul-retreat
tags: [gnoland, ecosystem, meetup, retreat, seoul]
authors: [michelleellen]
---

# Gnomes spotted in Seoul: Onbloc x Core Engineering Collaboration 

gno.land’s core engineering team has just wrapped up an intense and highly 
productive two-week coding sprint with our core contributor, 
[Onbloc](https://onbloc.xyz/). Hosted in Onbloc's offices in the heart of Seoul,
this onsite collaboration stands as one of our most successful development cycles
yet, setting new records for pull requests merged and advancing critical priorities
for our upcoming mainnet launch, especially with the GnoVM. 

[![office](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/thumbs/office.jpg)](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/office.jpg)

## Starting with a Bang: test5 and engineering developments

> The collaboration kicked off with the launch of test5. This milestone set the tone for two weeks of unparalleled productivity. During this sprint, the teams identified areas of collaboration like a feedback workshop on Gnoscan, Gnoswap, and Adena Wallet, and topics such as gas, realm upgradeability, gnoweb, tx-indexer, infrastructure, and ABCI queries. The benchmarking and gas cost estimations is a priority as the team aims to optimize transaction efficiency for both users and developers as we work our way towards mainnet. Realm development will advance significantly in the next few months, highlighting the need to enhance an existing realm's functionality and capabilities without creating a new one. To address this, a realm upgradeable pattern example is necessary to illustrate possible solutions to this challenge. This feature is particularly important for applications like Gnoswap, and was a key focus during the recent two-week sprint.

Author: ```suggestion
The collaboration kicked off with the launch of test5. This milestone set the tone for two weeks of unparalleled productivity. During this sprint, the teams identified areas of collaboration including a feedback workshop on Gnoscan, Gnoswap, and the Adena Wallet, and topics such as gas, realm upgradeability, `gnoweb`, `tx-indexer`, infrastructure, and ABCI queries. The benchmarking and gas cost estimations are a priority as the team aims to optimize transaction efficiency for both users and developers. Realm development will advance significantly in the next few months, highlighting the need to enhance an existing realm's functionality and capabilities without creating a new one. To address this, a realm upgradeable pattern example is necessary to illustrate possible solutions to this challenge. This feature is particularly important for applications like Gnoswap, and was a key focus during the recent two-week sprint.
```

---
publication_date: 2024-11-28T00:00:00Z
slug: seoul-retreat
tags: [gnoland, ecosystem, meetup, retreat, seoul]
authors: [michelleellen]
---

# Gnomes spotted in Seoul: Onbloc x Core Engineering Collaboration 

gno.land’s core engineering team has just wrapped up an intense and highly 
productive two-week coding sprint with our core contributor, 
[Onbloc](https://onbloc.xyz/). Hosted in Onbloc's offices in the heart of Seoul,
this onsite collaboration stands as one of our most successful development cycles
yet, setting new records for pull requests merged and advancing critical priorities

> for our upcoming mainnet launch.

Reviewer: I would throw in: `, especially for the GnoVM`.

[![office](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/thumbs/office.jpg)](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/office.jpg)

## Starting with a Bang: test5 and engineering developments

The collaboration kicked off with the launch of test5. This milestone set the
tone for two weeks of unparalleled productivity. During this time, the teams 
focused on benchmarking and gas cost estimation, aiming to optimize transaction
efficiency for both users and developers.

A significant priority was advancing `gnoweb`, gno.land’s universal web frontend. 
The team outlined a development [roadmap](https://github.com/gnolang/gno/issues/3191) 
for iterative feature releases, ensuring adaptability and a user-centric experience 
throughout the process. Key features include:
- Design Overhaul: Creating a more user-friendly UI with a color palette 
matching gno.land's true spirit.
- UX Redesign: Establishing `gnoweb` as the go-to realm web frontend.
- Code Refactoring: Enhancing security and maintainability.
- Navigation Improvements: Simplifying user navigation and introducing a query 
system for direct linking to specific `gnoweb` content or commands.

> Finally, the team collaborated closely with Onbloc’s DevOps experts to streamline 

Reviewer: We can add more things here, `gnoweb` was just one project:

https://hackmd.io/@JFGdLdZISka23j6-v8UYbA/H17Awcf7Jg/edit

We also helped Onbloc identify pain points in current Gnoswap, Adena and Gnoscan implementations, through a workshop with both teams

and automate the genesis creation and validation process for future testnets and 
the eventual mainnet launch.

## Strengthening Regional Community Connections

Beyond the engineering milestones, this sprint fortified relationships within
our contributor network and the broader community. Together with Onbloc, we 
hosted an informal meetup to engage with the Korean developer community and met
with Cosmostation, a longstanding South Korea-based Cosmos contributor. Over 15 
developers attended the event, eager to learn about gno.land’s vision and explore 
how they could contribute. The enthusiasm from local developers and the 
collaboration with Onbloc highlight the growing global momentum behind gno.land. 
To further support regional growth, we are prioritizing multilingual content,
starting with Korean. In collaboration with Onbloc, we created Korean video 
content for the community - coming soon! 

[![office](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/thumbs/meetup.jpg)](https://gnolang.github.io/blog/2024-11-28_seoul-retreat/src/meetup.jpg)

> ## A Celebration of Culture and Collaboration

Reviewer: This section is great 👌
